x-pack/legacy/plugins/reporting/server/browsers/safe_child_process.ts

```diff
@@ -50,17 +55,9 @@ export function safeChildProcess(childProcess: IChild, observer: InnerSubscriber
   // this is adding unsubscribe logic to our observer
   // so that if our observer unsubscribes, we terminate our child-process
   observer.add(() => {
+    logger.debug(`The browser process observer has unsubscribed. Closing the browser...`);
```

Comment:
I don't understand much of what this module is doing, but this line is one that I understand the least.

When does the callback get called? 

Also, this may have made sense with PhantomJS, but currently, [`childProcess` is a dummy object with a `kill` method that takes no parameters.](https://github.com/elastic/kibana/blob/4d871df45602b13a95478c8eb02c0e6f367e8284/x-pack/legacy/plugins/reporting/server/browsers/chromium/driver_factory/index.ts#L130) Not sure what we're hoping this code will do.

Reply:
Yeah, to be honest Chromium already runs inside of a child process: https://github.com/GoogleChrome/puppeteer/blob/master/lib/Launcher.js#L133. I don't think we need to wrap it in another process for safety reasons. Understandably, this is a bigger change to this file, so I'm not sure if that's something you want to tackle in this PR?

Reply:
haha, no way

Reply:
This sounds like fair game after we tackle some performance issues though